stock_location_orderpoint/models/stock_move.py

```diff
@@ -0,0 +1,87 @@
+from odoo import _, fields, models
+from odoo.tools import ormcache
+
+from odoo.addons.queue_job.job import identity_exact
+
+
+class StockMove(models.Model):
+    _inherit = "stock.move"
+
+    location_orderpoint_id = fields.Many2one(
+        "stock.location.orderpoint", "Stock location orderpoint"
+    )
+
+    @ormcache("self", "product")
+    def _get_location_orderpoint_replenishment_date(self, product):
+        return min(
+            self.filtered(lambda move: move.product_id == product).mapped("date")
+        )
+
+    def _prepare_location_orderpoint_replenishment(self, location_field, domain):
+        locations_products = {}
+        product_obj = self.env["product.product"]
+        for move in self:
+            if not move.filtered_domain(domain):
+                continue
+            location = getattr(move, location_field)
+            locations_products.setdefault(location, set())
+            locations_products[location].add(move.product_id.id)
+
+        for location, products in locations_products.items():
+            for product in product_obj.browse(products):
+                self._enqueue_auto_replenishment(
+                    location, product, location_field
+                ).delay()
+
+    def _enqueue_auto_replenishment(
+        self, location, product, location_field, **job_options
+    ):
+        """Enqueue a job stock.location.orderpoint.moves_auto_replenishment()
+
+        Can be extended to pass different options to the job (priority, ...).
+        The usage of `.setdefault` allows to override the options set by default.
+
+        return: a `Job` instance
+        """
+        job_options = job_options.copy()
+        job_options.setdefault(
+            "description",
+            _("Try to replenish quantities for location {} and product {}").format(
+                location.display_name, product.display_name
+            ),
+        )
+        # do not enqueue 2 jobs for the same location and product set
+        job_options.setdefault("identity_key", identity_exact)
+        delayable = self.env["stock.location.orderpoint"].delayable(**job_options)
+        job = delayable.run_auto_replenishment(
+            product,
+            location,
+            location_field == "location_id" and location_field or "location_src_id",
+        )
+        return job
+
+    def _action_confirm(self, *args, **kwargs):
+        """This triggers the replenishment for new moves which are waiting for stock"""
+        moves = super()._action_confirm(*args, **kwargs)
+        moves._prepare_location_orderpoint_replenishment(
+            "location_id",
+            self.env["stock.location.orderpoint"]._get_waiting_move_domain(),
+        )
+        return moves
+
+    def _action_done(self, *args, **kwargs):
+        """
+        This triggers the replenishment for waiting moves
+        when the stock increases on a source location of an orderpoint
```

Comment:
@mt-software-de In complement of my above comment, this does not do what it says.

All moves generate a job and not only for the ones that have an orderpoint !